### PR TITLE
fix(useMounted): The second argument in vue2.7 should be `undefined`

### PR DESCRIPTION
**Pull request by yanbowe**
### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

[#3803](https://github.com/vueuse/vueuse/issues/3803)

### CODE REVIEW

File: packages/core/useMounted/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function useMounted() {
   if (instance) {
     onMounted(() => {
       isMounted.value = true
-    }, isVue2 ? null : instance)
+    }, isVue2 ? void 0 : instance)
```
Comment (Glandos) [inline packages/core/useMounted/index.ts:16]: Isn't it better to use `undefined` and let bundlers do their job by using `void 0` if they want?

It's just my opinion, I prefer to read `undefined`, it's seems clearer to me.

Comment (Glandos) [inline packages/core/useMounted/index.ts:16]:
```suggestion
    }, isVue2 ? undefined : instance)
```

**Review by antfu — APPROVED** [review]

**Comment by susnux** [comment]
If there any plan for a new vueuse release with this?

**Comment by qnp** [comment]
Same, I'm looking forward to seeing this fix released, as I have errors that throw on my Vue2 project due to `null` target instead of `undefined`.

**Comment by r23qw** [comment]
@antfu 👋，is there any plan to release a version of this issue?

**Pull request merged**